Lib/test/test_asyncore.py

```diff
@@ -800,6 +801,36 @@ def test_quick_connect(self):
             if t.is_alive():
                 self.fail("join() timed out")
 
+    def test_map_altered_in_loop(self):
+        family = self.family
+        fail = self.fail
+
+        class NoWriteClient(BaseClient):
+
+            def handle_write(self):
+                fail("This is a non writable socket")
```

Comment:
I think we need to explain the issue in the error message, something like:

    Attempt to call handle_write on the wrong client